aten/src/ATen/core/autocast/register_autocast_ops.cpp

```diff
@@ -0,0 +1,528 @@
+*****************************************************************************************************************/
+
+auto register_fallthrough = c10::Dispatcher::singleton()
+  .registerBackendFallbackKernel(DispatchKey::AutocastTensorId,
+                                 KernelFunction::makeFallthrough());
+
+// TODO:  Codegen the stuff below?  Ed said
+// > you are going to have to write the function definition at some point, I wouldn't try to get clever about it
+// Therefore, for the moment, this is all copy pasted in from VariableTypeEverything.cpp with appropriate substitutions.
+
+// Macros to reduce boilerplate somewhat.
+// Common case where registration signature matches redispatch signature
+// (that's why SIGNATURE is repeated in the WrapFunction instantiation)
+#define KERNEL(FUNC, REGISTER_SCHEMA, SIGNATURE, POLICY) \
+  .op(torch::RegisterOperators::options() \
+    .schema(REGISTER_SCHEMA) \
+    .kernel<SIGNATURE>(DispatchKey::AutocastTensorId, \
+    &WrapFunction<CastPolicy::POLICY, SIGNATURE, SIGNATURE, FUNC>::type::call) \
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+
+#define KERNEL_UNBOXED_ONLY(FUNC, REGISTER_SCHEMA, SIGNATURE, POLICY) \
+  .op(torch::RegisterOperators::options() \
+    .schema(REGISTER_SCHEMA) \
+    .impl_unboxedOnlyKernel<SIGNATURE, \
+    &WrapFunction<CastPolicy::POLICY, SIGNATURE, SIGNATURE, FUNC>::type::call \
+    >(DispatchKey::AutocastTensorId) \
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+
+// Less-common but still useful case: redispatching to a function with a new signature (e.g. appending a dtype)
+#define KERNEL_UNBOXED_ONLY_DIFFERENT_REDISPATCH_SIGNATURE(REDISPATCH_FUNC, REGISTER_SCHEMA, REGISTER_SIGNATURE, REDISPATCH_SIGNATURE, POLICY) \
+  .op(torch::RegisterOperators::options() \
+    .schema(REGISTER_SCHEMA) \
+    .impl_unboxedOnlyKernel<REGISTER_SIGNATURE, \
+    &WrapFunction<CastPolicy::POLICY, REGISTER_SIGNATURE, REDISPATCH_SIGNATURE, REDISPATCH_FUNC>::type::call \
+    >(DispatchKey::AutocastTensorId) \
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+
+/*****************************************
+Explicit registration for out-of-place ops
+*****************************************/
+auto register_out_of_place = torch::RegisterOperators()
+  // fp16
+  // TODO (to consider): should the slow_conv*d from python_nn_functions_dispatch.h go here?
+  // They use unfolding + admm (which may use cublas) so it's possible they benefit from Tensor Cores,
+  // although if that code path is being used, it's probably unsalvageably slow regardless.
+  KERNEL_UNBOXED_ONLY(at::_convolution, "aten::_convolution(Tensor input, Tensor weight, Tensor? bias, int[] stride, int[] padding, int[] dilation, bool transposed, int[] output_padding, int groups, bool benchmark, bool deterministic, bool cudnn_enabled) -> Tensor", Tensor (const Tensor &, const Tensor &, const Tensor &, IntArrayRef, IntArrayRef, IntArrayRef, bool, IntArrayRef, int64_t, bool, bool, bool), fp16)
```

Comment:
More thinking out loud (not actionable for this PR): In an ideal world, I feel like you should just be able to write `KERNEL_UNBOXED_ONLY(at::_convolution, fp16)` and that would be enough information (we'd need separate function names for each overload, in the case of overloads, I think).

Reply:
Yeah, overloads are why I need the function name AND the full signature in each case.